controllers/main.js

```diff
@@ -227,6 +231,7 @@ module.exports.updatePlayData = async (req, res, next) => {
       { sessionId },
       { $set: { sessionEnded: true, sessionScore: score } }
     );
+    if(currentUser){
```

Comment:
This currentUser logic should probably be moved higher, before the rest of the logic is executed to improve performance and efficiency